preprocess/process_dicom.py

```diff
@@ -129,28 +129,31 @@ def rescale_dicom_image_intensity(image, meta_dict):
         # '0028|1050' is the DICOM window Width for display (string). It is
         # used to apply an intensity windowing filter.
         width = int(meta_dict['0028|1051'])
+        intensity_winwdow_filter = itk.IntensityWindowingImageFilter.New(image)
+        # ITK versions prior to ITK 5.0.0 beta 2 do not support passing tuples
+        # directly as an argument of the `New()` function. Instead, we need to
+        # explicitely call the correct `Set` function.
+        intensity_winwdow_filter.SetWindowLevel(width, center)
+        # Image is rescaled between 0 and 255 to be saved as a JPEG later.
+        intensity_winwdow_filter.SetOutputMinimum(0)
+        intensity_winwdow_filter.SetOutputMaximum(255)
+        intensity_winwdow_filter.Update()
+        return intensity_winwdow_filter.GetOutput()
     except KeyError:
-        raise Exception("DICOM window `center` or `width` tag not found.")
-
-    intensity_winwdow_filter = itk.IntensityWindowingImageFilter.New(image)
-    # ITK versions prior to ITK 5.0.0 beta 2 do not support passing tuples
-    # directly as an argument of the `New()` function. Instead, we need to
-    # explicitely call the correct `Set` function.
-    intensity_winwdow_filter.SetWindowLevel(width, center)
-    intensity_winwdow_filter.Update()
-    return intensity_winwdow_filter.GetOutput()
+        # If either or both of the required tags are missing in the DICOM,
+        # we simply rely on the image minimum and maximum to rescale it.
+        # Image is rescaled between 0 and 255 to be saved as a JPEG later.
+        return itk.RescaleIntensityImageFilter(image, OutputMinimum=0,
+                                               OutputMaximum=255)
```

Comment:
An output image explicitly specified to have a pixel type of `unsigned char` will ensure the correct output type and will automatically specify `OutputMinimum` and `OutputMaximum`.

Reply:
True, but I was on purpose not converting the pixel type to `unsigned char` at this point of the script, before resampling, to keep all the image values as long as possible (until the last step).